### DISCUSSION TITLE
[Feature request] Ban messages with button links

### DISCUSSION THREAD

**Discussion by ShGKme** [umputun/tg-spam#131] [Ideas]
Some spam bots makes an image-only message with a button-links.

- Real users often send image-only messages
- Real users often send links

So both image-only and links filters may not be applicable.

However, only spam-bots uses **button-links**.

This would be nice to have a new filter for that kind of messages.

Example:

![image](https://github.com/user-attachments/assets/a189dd90-5614-499a-9ad8-39ff17a95be4)

**Comment by umputun** [comment]
I have never seen such spam messages. If it can be detected by either image-only or links-only, I don't see why we shouldn't use it. Normal users who send the first message as a picture without text or just a link are already a red flag, and it makes sense to consider them potential spammers.

Anyway, I'll transfer it to discussions/ideas, and if anyone wants to submit a PR, I'll consider it.

**Reply by ShGKme** [reply]
> I have never seen such spam messages

We had it 5 times yesterday. Fortunately it was a scam-job spam. In past there were similar with cp 🙈.

> Picture without text or just a link are already a red flag

This is a very common type of messages for us. People ask for help. They send an image of a code or an error. They send a link to a sandbox with reproduction, or others reply with a link to a documentation. Just yesterday we had 18 images sent by users.

<details>

<summary>Example of an image sent as a second message from a new user</summary>

![image](https://github.com/user-attachments/assets/5f81ab12-7313-480d-9f78-6807ecf5c8ed)

</details>

We have several developers communities up to 10k+ members each and about 120-150 spam bots per day. So it's important to have false-positive as low as possible. Even 0.5% - 1% of false detection means one banned real user everyday.

**Reply by umputun** [reply]
I'm not arguing about the nature of messages in your groups; it may be fine to receive the first N messages that contain only pictures or links. In my groups, I check only the first message, and receiving a first message as a picture without any text is very unusual. Marking those as spam has worked well in my cases. I think I've encountered a single false positive due to this check.

Anyway, I have nothing against this check. As I mentioned above, I would consider PR implementing the check if it doesn't complicate the code too much.
